### PR TITLE
gensim.prepare speed up

### PR DESCRIPTION
**Pull request by mattilyra**
*This is PR is unfinished*

As it stands the `gensim.prepare` contains a lot of unnecessary iteration over lists of tuples. The aim here is to make the whole thing run faster, but this needs more testing.

I am not entirely sure if all the indexing and ordering of the vocabulary items is correct - I tested it on one lda model but I think this needs a bit more testing. I wanted to get some feedback on the PR though before I test it further.

### CODE REVIEW

**Comment by bmabey** [comment]
Great, thanks!

At one point I was using `topic_model.state.get_lambda()` but I swapped it out because I was told that it wasn't returning the matrix in the same order as my terms list. Have you verified that the ordering is correct?

**Comment by mattilyra** [comment]
It'll be in a different order than the term list but I think you can reorder the matrix using the indices from `dictionary.token2id`. The `lambda` matrix should be in order by `term ID`s where as the terms as they are nor are in whatever order `dictionary.token2id.keys()` gives you, but the `id` corresponds to the columns of lambda (as far as I understand).

I tested this relatively quickly on the data I was using and it seems to be doing the correct thing (the visualisation is the same using either method) but that still needs to be tested more thoroughly. It would certainly be faster as calling `topic_model.show_topics` does a couple of loops around the corpus internally plus at least one externally.

I mainly wanted to check if it's worth me putting my time into testing this.

**Comment by bmabey** [comment]
Yeah, the improvement is certainly welcomed and I will merge it in once we are confident it works as the previous version on master.  Are you using python 2 or 3 to test this?

**Comment by mattilyra** [comment]
I'm currently on 3 but switching to 2 isn't a problem. I guess it should be tested on both?

**Comment by bmabey** [comment]
Yeah, we want to support both but I never put a unit test in for the gensim stuff so it has (or hasn't!) been tested manually. Adding a simple smoke test to see if it blows up or not would be sufficient for testing it in both I think. Or you can do it manually.

**Comment by mattilyra** [comment]
Ok, I've built two visualisations one with the `master` and one with code in this PR. Both models look the same to me. The LDA model is the same from the example notebook (http://nbviewer.ipython.org/github/bmabey/pyLDAvis/blob/master/notebooks/Gensim%20Newsgroup.ipynb), I have the visualisations saved as static HTML which I put into a separate branch (https://github.com/mattilyra/pyLDAvis/tree/html_vis) - I won't do a PR on that so as not to add crud to master, but you can check them out and verify for yourself - to me they look the same (lda.2.html is the one from this PR).

For the smallish model built from 20 newsgroups data the code runs about 15 seconds faster than the orignal code, that's about a 25% increase in speed.

**Comment by bmabey** [comment]
Awesome work @mattilyra, thanks! Is this ready to be merged in then?

File: pyLDAvis/gensim.py [hunk 1]
```diff
@@ -10,46 +10,72 @@
 from past.builtins import xrange
 from . import prepare as vis_prepare
```
Comment (utasthana) [inline pyLDAvis/gensim.py]: where is . package

**Pull request merged**